cherrypy/process/servers.py

```diff
@@ -185,7 +185,7 @@ def _get_base(self):
         if not self.httpserver:
             return ''
         host, port = self.bind_addr
-        if getattr(self.httpserver, 'ssl_adapter', None):
+        if getattr(self,'ssl_certificate', None):
```

Comment:
since the code was failing for certain 3.x versions of python. That check is needed for running the alternative code for python versions greater than 2 (i.e. 3.x). 
Instead of six.PY2 or six.PY3 , sys.version_info can also be used to find the current version of python.
And I didnt know what new tests to add. I just put that check to fix the error in 3.x versions.